js/textext.plugin.moztrap-ajax.js

```diff
@@ -0,0 +1,255 @@
+/**
+ * jQuery TextExt Plugin
+ * http://textextjs.com
+ *
+ * @version 1.3.1
+ * @copyright Copyright (C) 2011 Alex Gorbatchev. All rights reserved.
+ * @license MIT License
+ *
+ * This is a modified plugin for MozTrap REST API
+ */
+(function($)
+{
+	function TextExtAjax() {};
+
+	$.fn.textext.TextExtAjax = TextExtAjax;
+	$.fn.textext.addPlugin('ajax', TextExtAjax);
+
+	/*
+	 * this var is used to mapping 'Modifier" to a correct Query name for MozTrap REST API
+	 * TODO: 1. change 'status' to 3 default results (active/draft/disable)
+	 */
+	var searchTypeMapModifiers = {
+		"caseversion" : {
+			"name" : "caseversion",
+		 	"tag" : "tag",
+			"suite" : "suite",
+			"product" : "product",
+			"ver" : "productversion",   //still get 'undefined' results sometimes
+			"status" : "caseversion"
+		},
+		"suite" : {
+			"name" : "suite",
+			"product" : "product",
+			"status" : "suite"
+		},
+	};
+
+	var p = TextExtAjax.prototype,
+
+
+		OPT_DATA_CALLBACK = 'ajax.data.callback',
+		
+		OPT_CACHE_RESULTS = 'ajax.cache.results',
+		
+		OPT_LOADING_DELAY = 'ajax.loading.delay',
+
+		OPT_LOADING_MESSAGE = 'ajax.loading.message',
+
+		OPT_TYPE_DELAY = 'ajax.type.delay',
+
+
+
+		EVENT_SET_SUGGESTION = 'setSuggestions',
+
+		EVENT_SHOW_DROPDOWN = 'showDropdown',
+
+		TIMER_LOADING = 'loading',
+
+		DEFAULT_OPTS = {
+			ajax : {
+				typeDelay      : 0.5,
+				loadingMessage : 'Loading...',
+				loadingDelay   : 0.5,
+				cacheResults   : false,
+				dataCallback   : null
+			}
+		}
+		;
+	p.init = function(core)
+	{
+		var self = this;
+
+		self.baseInit(core, DEFAULT_OPTS);
+		self.on({
+			getSuggestions : self.onGetSuggestions
+		});
+
+		/*
+		 * use 'ajax.params' to pass required info
+   		 */
+		self._suggestions = null;
+		self._url = self._core._opts.ajax.params.url + "/api/v1/" || "http://moztrap.mozilla.org/api/v1/";
+		self._searchType = self._core._opts.ajax.params.searchType || "caseversion";
+		self._listLimit = self._core._opts.ajax.params.listLimit || 20;
+		self._modifier = null;
+	};
+
+	p.load = function(query)
+	{
+		var self = this,
+		    dataCallback = self.opts(OPT_DATA_CALLBACK) || function(query) { return { q : query } },
+		    opts,
+		    url,
+		    tmpQuery = searchTypeMapModifiers[self._searchType][self._modifier] || "unkown"
+		    ;
+
+		tmpQuery = (tmpQuery === "unkown") ? "caseversion" : tmpQuery;  //default set to caseversion?
+                url = self._url + tmpQuery + "/?limit=" + self._listLimit;
+
+		opts = $.extend(true,
+			{
+				url     : url,                              //point to moztrap API url
+				data    : dataCallback(query),
+			        //TODO: support different response type
+				success : function(data) { self.onComplete(data, query) },
+				error   : function(jqXHR, message) { console.error(message, query) }
+			}, 
+			self.opts('ajax')
+		);
+	        //TODO: initiate multiple call to all REST APIs
+		$.jsonp(opts); //TODO: maybe change this with jsonp?
```

Comment:
We have already removed MozTrap API's CORS restrictions. I guess we can just use `$.ajax()` instead.